staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go

```diff
@@ -213,6 +241,20 @@ func (s *Serializer) Decode(originalData []byte, gvk *schema.GroupVersionKind, i
 		return nil, actual, err
 	}
 
+	// Handle strict mode.
+	if s.strict {
+		// Delayed failure for the strict YAML to JSON conversation,
+		// so that the error output also includes GVK.
+		if errorYAMLToJSON != nil {
+			return nil, actual, fmt.Errorf("%#v: %v", actual, errorYAMLToJSON)
+		}
+		// Return an error for unknown JSON fields in strict mode.
+		strictObj := obj.DeepCopyObject()
+		if err := yaml.UnmarshalStrict(data, strictObj); err != nil {
```

Comment:
if we were going to decode twice, I didn't expect to decode into an already-populated object, but a new empty object

Reply:
i need to figure out how/where to make the copy, if we are going to copy even.

Reply:
@neolit123 you might be able to use `scheme.New` for allocating a new, empty object of the same gvk as well

Reply:
> note that the strictObj could also be a fresh object by passing nil object into UseOrCreateObject.

thanks for the tip.

> @neolit123 you might be able to use scheme.New for allocating a new, empty object of the same gvk as well

hm, do we even have a scheme here passed to the function.